jaq-core/src/lib.rs

```diff
@@ -402,3 +404,82 @@ const LOG: &[(&str, usize, RunPtr, UpdatePtr)] = &[(
     |_, cv| box_once(Ok(debug(cv.1))),
     |_, cv, f| f(debug(cv.1)),
 )];
+
+#[cfg(feature = "format")]
+fn stringify(v: Val) -> String {
+    v.clone()
+        .to_str()
+        .map_or_else(|_| v.to_string(), |s| s.to_string())
+}
+
+#[cfg(feature = "format")]
+fn to_csv(vs: &[Val], delimiter: u8) -> ValR {
+    use csv::{StringRecord, WriterBuilder};
+    let mut writer = WriterBuilder::new()
+        .delimiter(delimiter)
+        .from_writer(vec![]);
+    let record = StringRecord::from(
+        vs.iter()
+            .map(|cell| stringify(cell.clone()))
+            .collect::<Vec<_>>(),
+    );
+    writer.write_record(&record).map_err(Error::from_any)?;
+    Ok(Val::str(
+        String::from_utf8(writer.into_inner().map_err(Error::from_any)?)
+            .map_err(Error::from_any)
+            .map(|s| s[0..s.len() - 1].to_string())?,
+    ))
+}
+
+#[cfg(feature = "format")]
+const FORMAT: &[(&str, usize, RunPtr)] = &[
+    ("@text", 0, |_, cv| box_once(Ok(Val::str(stringify(cv.1))))),
+    ("@json", 0, |_, cv| box_once(Ok(Val::str(cv.1.to_string())))),
+    ("@html", 0, |_, cv| {
+        box_once(Ok(Val::str(
+            html_escape::encode_safe(&stringify(cv.1)).to_string(),
+        )))
+    }),
+    ("@uri", 0, |_, cv| {
+        box_once(Ok(Val::str(
+            urlencoding::encode(&stringify(cv.1)).to_string(),
+        )))
+    }),
+    ("@urid", 0, |_, cv| {
+        box_once(
+            urlencoding::decode(&stringify(cv.1))
+                .map_err(Error::from_any)
+                .map(|s| Val::str(s.to_string())),
+        )
+    }),
+    ("@csv", 0, |_, cv| {
+        box_once(cv.1.as_arr().and_then(|vs| to_csv(vs, b',')))
```

Comment:
Can you add a test that shows what happens when the input contains commas (inside a string)?

Reply:
Thanks!